src/unstract/adapters/x2text/llm_whisperer/src/llm_whisperer.py

```diff
@@ -320,3 +309,49 @@ def process(
             logger.error(f"OS error while writing {output_file_path}: {e} ")
             raise ExtractorError(str(e))
         return output
+
+    def process(
+        self,
+        input_file_path: str,
+        output_file_path: Optional[str] = None,
+        **kwargs: dict[Any, Any],
+    ) -> str:
+        """Used to extract text from documents.
+
+        Args:
+            input_file_path (str): Path to file that needs to be extracted
+            output_file_path (Optional[str], optional): File path to write
+                extracted text into, if None doesn't write to a file.
+                Defaults to None.
+
+        Returns:
+            str: Extracted text
+        """
+        response: requests.Response = self._send_whisper_request(input_file_path)
+
+        return self._process_extract_text_from_response(output_file_path, response)
+
+    def process_with_hash(
+        self,
+        input_file_path: str,
+        output_file_path: Optional[str] = None,
+        **kwargs: dict[Any, Any],
+    ) -> dict[str, str]:
+        """Used to extract text from documents.
+
+        Args:
+            input_file_path (str): Path to file that needs to be extracted
+            output_file_path (Optional[str], optional): File path to write
+                extracted text into, if None doesn't write to a file.
+                Defaults to None.
+        Returns:
+            dict[str, str]: extrcted text, whisper_hash
+        """
+        output = {}
+
+        response: requests.Response = self._send_whisper_request(input_file_path)
+        output["extracted_text"] = self._process_extract_text_from_response(
+            output_file_path, response
+        )
+        output["whisper_hash"] = response.headers.get("whisper-hash", "")
+        return output
```

Comment:
1. This would return a dictionary - how do we plan to use this function?  If I understand right, the requirement is to return the whisper-hash for all extractions right - why do we need 2 separate functions in this case?
1. Also how will this concrete method be used in all places where extraction is done? There's no `whisper-hash` present for other extractors right?

My suggestion would be to maybe modify the contract for `process()` itself to return maybe the extracted text and some metadata on it. The `whisper-hash` can be a field within this metadata, later we could have cost / latency / other info if necessary

Reply:
@johnyrahul I think I have the same question as Chandru as to where are we calling this method today? Also  backing the thought that we should build a data structure around the extraction response if we see additional attributes coming in.

Reply:
@chandrasekharan-zipstack @gaya3-zipstack  does this PR makes sense

https://github.com/Zipstack/unstract-sdk/pull/64

Reply:
@johnyrahul my concern is that we have `LLMWhisperer` adapter alone that has a new method `process_with_hash()` - since this is not generic / added to the base class we'll end up with client code such as
```
if x2text_adapter_type == "LLMWhisperer":  # provided as an example, might not be a valid check
   x2text_inst.process_with_hash()
else:
    x2text_inst.process()
```
I'm not a huge fan of this and I'd instead vote to update the contract to return as a JSON maybe. However I understand that if its a priority / we wish to avoid a breaking change now, I can approve this PR

Reply:
@johnyrahul Folks, agree here that we do not need a new function ideally.

Reply:
@chandrasekharan-zipstack agrees with your comment here.  I will update the contract to return JSON.
As we will be updating SDK also, I guess we should be able to handle the breaking changes along with SDK versioning

Reply:
Still, we have to extract the response conditionally on the upstream side.